csv_joiner.py

```diff
@@ -0,0 +1,147 @@
+import sys
+
+
+class InvalidModeError(Exception):
+    def __init__(self, modes):
+        self.expected = '/'.join(modes)
+        self.message = 'Invalid join mode'
+
+    def __str__(self):
+        return f'{self.message}. Expected: {self.expected}'
+
+
+class InvalidColumnError(Exception):
+    def __init__(self, col):
+        self.col = col
+        self.message = 'Column "{0}" does not appear in one/both files'
+
+    def __str__(self):
+        return self.message.format(self.col)
+
+
+class Parser:
+    @staticmethod
+    def parse_line(line: str) -> list:
+        line = line.rstrip()
+        start_i = 0
+        items = []
+        quotes = 0
+        for i, char in enumerate(line):
+            if char == ',' and not quotes:
+                items.append(line[start_i:i])
+                start_i = i + 1
+
+            if char == '"':
+                quotes += 1
+
+            try:
+                if line[i + 1] == ',' and quotes % 2 == 0:
+                    quotes = 0
+            except IndexError:
+                break
+
+        items.append(line[start_i:])
+        return items
+
+
+class Line:
+    @staticmethod
+    def add_to_dict(list_of_items: list, keys_dict: dict):
+        new_dict = {}
+        for item, key in zip(list_of_items, keys_dict.keys()):
+            new_dict[key] = item
+        return new_dict
+
+    @staticmethod
+    def print_line(merged_dict: dict, order: list):
+        out_line = ''
+        for col in order:
+            out_line += ',' + merged_dict[col]
+        print(out_line[1:])
+
+
+class Header:
+    order = []
+
+    def __init__(self, line):
+        self.line = line
+        self.list = Parser.parse_line(self.line)
+        self.dict = self.list_to_dict()
+
+    def list_to_dict(self):
+        headers_dict = {}
+        for i in self.list:
+            headers_dict[i] = 'None'
+        return headers_dict
```

Comment:
Should this be done in the `Header` class? maybe the caller should have a separate method or there should be another abstraction?